### ISSUE TITLE
Checkout: Add PayPal payment option on Subscribe page if there is saved PayPal payment information

### ISSUE THREAD

**Issue by data-sync-user** [mozilla/fxa#7098]
As a simplification, for the MVP, we would like to only allow one payment option at a time. Therefore, similar to paying with Stripe, if we have saved PayPal payment information, we should have a screen to enable the user to subscribe to a product using it.

Edit: Also once #7115 is complete, this patch should add the logic to check whether an FxA user is an existing PayPal customer with saved payment information.

Open question(s):

* If the payment information is invalid, expired or otherwise unusable, do we show an error message here as we do for subscription management?

Task:

* Use the new Customer object to get the PayPal properties we need for information.
* Update logic for what components to show based on new Customer object properties
* Create new front end components for PayPal UI
* Write tests for those components in Jest.

This should be much easier after FXA-2800; we just have to find out what billing information we can/should show.

┆Issue is synchronized with this [Jira Task](https://jira.mozilla.com/browse/FXA-2801)
┆Issue Number: FXA-2801

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

> This should be much easier after FXA-2800; we just have to find out what billing information we can/should show.

Since there can be many different funding sources for PayPal ( https://developer.paypal.com/docs/business/javascript-sdk/javascript-sdk-configuration/#disable-funding ) (see also FXA-2783 ( https://jira.mozilla.com/browse/FXA-2783 )), in contrast to Stripe which is only credit card, the recommendation is to just show that their billing source is PayPal.

**Comment by data-sync-user** [comment]
➤ Jessica Cook commented:

Hi [~bchen@mozilla.com] adding the link to the saved payment option spec here https://www.figma.com/file/ejDMYaKpm5bsYL5YvhItur/Payment-Integration?node-id=696%3A4

**Comment by data-sync-user** [comment]
➤ Barry Chen commented:

A few notes before I set this back to To Do.

The issue is currently blocked by FXA-3069.

There's a WIP branch at https://github.com/chenba/fxa/tree/2801-paypal-already.  That branch is based on the work Lesley did in FxA PR 7570, which is not yet in main.   ( https://github.com/chenba/fxa/tree/2801-paypal-already. )There a few failing existing tests in the branch; no new tests so far.  And form submission callback to call the API to create the additional subscription is not yet implemented.

**Comment by data-sync-user** [comment]
➤ Bianca Danforth commented:

Thanks for the update Barry. Would it not be possible to agree on a route and a request/response object and land a patch? That would require FXA-3069 to handle any integration fixes, I guess.

**Comment by data-sync-user** [comment]
➤ Barry Chen commented:

Short answer: sure, definitely.

Long answer... well first I need PR #7570 to go in first since my branch contains that work already.  And unless the work for this issue is blocking some other work, I don't see merging it ahead of FXA-3069 will gain us any functionality.  (It will get us some completed points on the progress bar I guess.)

Personally I like to get this in quickly so I don't have to deal with merge conflicts.  😛

**Comment by data-sync-user** [comment]
➤ Barry Chen commented:

I'm going to set this back to To Do since we are descoping FXA-3078 for the initial PayPal release.  Status update below.

My WIP branch (https://github.com/chenba/fxa/tree/2801-paypal-already) is up to date with `main` at the time of writing.  The UI displays the correct saved payment option.   The button to pay is disabled since we need FXA-3069 in order to process the subscription.

**Comment by data-sync-user** [comment]
➤ Barry Chen commented:

Removing the Sprint since this has been descoped for the MVP.

**Comment by data-sync-user** [comment]
➤ Valentina Virlics commented:

QA verified here the foloswing scenarios:

  1. Have a paypal subscription for VPN, add a second subscription for Firefox Variety (or other) - US - should work
  2. Have a paypal subscription for VPN, add a second subscription for 123Done...Pro (or other) - EU - should work
  3. Have a paypal subscription for VPN US, add a second subscription for 123Done...ProEU (or other) - should not work
  4. Have a canceled paypal subscription for VPN (be in the availability period), add a second subscription for Firefox Variety (or other) - should work
  5. Have an expired paypal subscription for VPN (the subscription is no longer displayed in manage subscription page), add a (second) subscription for Firefox Variety (or other) - should work
  6. Have an invalid (a red banner in Manage Subscription page) paypal subscription for VPN, add a second subscription for Firefox Variety (or other) - should not work

All these secenarios work as expected. For 3 and 6 the "Something went wrong. Try again later" error is displayed after clicking on the "Pay with PayPal" button.
For 6 we should improve the message as there is no longer a rare scenario as is for 3. (https://jira.mozilla.com/browse/FXA-3395)

If anything else need checking here, please let us know, until then removing the "needs-qa" label.
